src/OrchardCore.Modules/OrchardCore.Layers/Services/DefaultLayersMethodProvider.cs

```diff
@@ -50,6 +51,16 @@ public DefaultLayersMethodProvider()
                 })
             };
 
+            _isInRole = new GlobalMethod
+            {
+                Name = "isInRole",
+                Method = serviceProvider => (Func<string, bool>) (role =>
+                {
+                    var httpContext = serviceProvider.GetRequiredService<IHttpContextAccessor>().HttpContext;
+                    return httpContext.User?.IsInRole(role) == true;
```

Comment:
I am not a big fan of `== true`.

Reply:
It's needed because the `?` after `User` makes it a `bool?` instead of a `bool`.

Reply:
What about 

```
httpContext.User?.IsInRole(role) ?? false;
```

Reply:
Makes it less readable imho. Also, `== true` is in line with the other rules. 

https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore.Modules/OrchardCore.Layers/Services/DefaultLayersMethodProvider.cs#L49

Reply:
There are already defined some liquid filters in Users module. We could add the scripting functions in that module, be in parity with what provided for users in liquid and even reuse some code. see https://github.com/OrchardCMS/OrchardCore/blob/54658bc8a427bbe1f6f3eb1f9f325a38204b3849/src/OrchardCore.Modules/OrchardCore.Users/Liquid/IsInRoleFilter.cs#L11

Reply:
> If User is null, I think that the httpContext.User?.IsInRole(role) will throw NRE

No, the `?` handles that.

Reply:
I agree with @MichaelPetrinolis on the NRE. `httpContext.User?.IsInRole()` is not the same as `httpContext?.User?.IsInRole`

Reply:
@sebastienros @MichaelPetrinolis User can be `null` but isn't it safe to assume httpContext is never `null` in OC? If not, I need to update `isAnonymous` and `isAuthenticated` too.